Sources/Changeset.swift

```diff
@@ -34,7 +42,7 @@ It detects additions, deletions, substitutions, and moves. Data is a `Collection
 
   - seealso: `Changeset.editDistance`.
 */
-public struct Changeset<T: Collection> where T.Iterator.Element: Equatable, T.IndexDistance == Int {
+public struct Changeset<T: Collection> where T.Iterator.Element: Equatable, T.IndexDistance == Edit<T.Iterator.Element>.Offset {
```

Comment:
It just occurs to me that arguably the collection's `IndexDistance` type would be even more correct than to use `Int` as the `Offset` type. Consider [the definition of `IndexDistance`](https://developer.apple.com/documentation/swift/collection/1781783-indexdistance):

> A type that represents the number of steps between a pair of indices.

This is exactly what we are trying to express with `Offset`: the number of steps between the collection's `startIndex` and the offset.

While `Int` is the default type for `IndexDistance` and almost all Swift collections use `Int`, there are some exceptions. For example, the `IndexDistance` type of `AnyCollection` is:

```swift
typealias AnyCollection<Element>.IndexDistance = IntMax
```

which is `IntMax == Int64 == Int` on 64-bit systems but not on 32-bit systems. This means that one currently cannot create a `Changeset<AnyCollection<T>>` on a 32-bit system because the constraint `T.IndexDistance == Edit<T.Iterator.Element>.Offset` would be violated. It's not a big deal in practice, but it is a bit unelegant.

What could we do about it? One option would be to make `Edit` also generic over the offset type:

```swift
public struct Edit<T: Equatable, Offset: SignedInteger> {
    // We don't need the typealias anymore
    // DELETE: public typealias Offset = Int
    ...
}

// And below:
// No IndexDistance constraint
public struct Changeset<T: Collection> where T.Iterator.Element: Equatable {
    ...
    // But all Edit types vended by Changeset now look like this:
    public let edits: [Edit<T.Iterator.Element, T.IndexDistance>]
    ...
    public static func edits(from source: T, to target: T) -> [Edit<T.Iterator.Element, T.IndexDistance>] {
```

Another option would be to make `Edit` generic over the collection instead of the element type:

```swift
public struct Edit<C: Collection> where C.Iterator.Element: Equatable {
    public typealias Element = C.Iterator.Element
    public typealias Offset = C.IndexDistance
    ...
}
```

This would probably mean that you couldn't easily compute a Changeset between two different collection types anymore (say, an `[Character]` and a `String.CharacterView`, but the current API doesn't currently support that anyway. It's probably not a big deal.

There might be other consequences that I haven't thought of, though. In any case, something like `Edit<[Int]>` does look a bit more complicated that `Edit<Int>`, although the more I think about it, the more "correct" the first version feels to me. After all, an `Edit` is really an operation on a collection – it doesn't make sense to view it independently of a collection.

I'm not sure it's worth it, though. It's certainly a much bigger change that the renaming using a typealias (and it's really a different issue).